msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/dcf/AbstractSignInFromOtherDeviceTest.java

```diff
@@ -0,0 +1,99 @@
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.msal.automationapp.testpass.broker.dcf;
+
+import androidx.annotation.NonNull;
+
+import com.microsoft.identity.client.Prompt;
+import com.microsoft.identity.client.msal.automationapp.R;
+import com.microsoft.identity.client.msal.automationapp.sdk.MsalAuthTestParams;
+import com.microsoft.identity.client.msal.automationapp.sdk.MsalSdk;
+import com.microsoft.identity.client.msal.automationapp.testpass.broker.AbstractMsalBrokerTest;
+import com.microsoft.identity.client.ui.automation.TokenRequestTimeout;
+import com.microsoft.identity.client.ui.automation.interaction.microsoftsts.AadLoginComponentHandler;
+import com.microsoft.identity.labapi.utilities.client.LabQuery;
+import com.microsoft.identity.labapi.utilities.constants.AzureEnvironment;
+import com.microsoft.identity.labapi.utilities.constants.TempUserType;
+import com.microsoft.identity.labapi.utilities.constants.UserType;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+// Brokered Auth verify "Sign In from other device" option.
+//@RetryOnFailure(retryCount = 2)
+abstract class AbstractSignInFromOtherDeviceTest extends AbstractMsalBrokerTest {
+
+    private final AzureEnvironment mAzureEnvironment;
+
+    public AbstractSignInFromOtherDeviceTest(@NonNull AzureEnvironment environment) {
+        mAzureEnvironment = environment;
+    }
+
+    @Override
+    public LabQuery getLabQuery() {
+        return LabQuery.builder()
+                .userType(UserType.CLOUD)
+                .azureEnvironment(mAzureEnvironment)
+                .build();
+    }
+
+    @Override
+    public TempUserType getTempUserType() {
+        return null;
+    }
+
+    @Override
+    public String[] getScopes() {
+        return new String[]{"User.read"};
+    }
+
+    @Override
+    public String getAuthority() {
+        return mApplication.getConfiguration().getDefaultAuthority().getAuthorityURL().toString();
+    }
+
+    protected void testSignInFromOtherDevice() throws Throwable {
+        final MsalSdk msalSdk = new MsalSdk();
+        final List<Map.Entry<String, String>> extraQP = new ArrayList<>();
+        extraQP.add(new AbstractMap.SimpleEntry<>("is_remote_login_allowed", Boolean.toString(true)));
+
+        final MsalAuthTestParams authTestParams = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .scopes(Arrays.asList(mScopes))
+                .msalConfigResourceId(getConfigFileResourceId())
+                .extraQueryParameters(extraQP)
+                .build();
+
+        msalSdk.acquireTokenInteractiveAsync(authTestParams, () ->
```

Comment:
So you don't wait for the test to complete? You are only interested in being navigated to a particular url?

Reply:
If i remember correctly i think i had trouble reading the device code when testing this flow in ui automation in the past, so i had to stop my test short (see `TestCase2110359.java`). I'll let @mohitc1 confirm though

Reply:
yes, only reading till the URL. From there it goes outside of the device.

Reply:
> If i remember correctly i think i had trouble reading the device code when testing this flow in ui automation in the past, so i had to stop my test short (see `TestCase2110359.java`). I'll let @mohitc1 confirm though

I was able to verify till last screen which displays the URL.